src/main/java/org/datavyu/util/MediaTimerTask.java

```diff
@@ -0,0 +1,46 @@
+package org.datavyu.util;
+
+import org.datavyu.plugins.ffmpeg.MediaPlayer;
+import org.datavyu.plugins.ffmpeg.NativeMediaPlayer;
+
+import java.lang.ref.WeakReference;
+import java.util.Timer;
+import java.util.TimerTask;
+
+public class MediaTimerTask extends TimerTask {
+    private Timer mediaTimer = null;
+    public static final Object timerLock = new Object();
```

Comment:
Can we lock on the media timer task itself instead?

Reply:
Why do we have this lock?

Reply:
We can't use the instance of the class to synchronize because we need to use the lock outside of the class, the lock is used to ensure that only one player at a time can call the update time method.

Reply:
Why not then put the lock round the update mechanism inside the update method?

Using locks is sometimes unavoidable... but exposing them is a bad design (yes, we have that in the c++ code which we inherited from ffplay).

Reply:
Removed the lock, and removed the MediaTimerTask as well